libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -215,10 +217,10 @@ private boolean validLocationUpdate(Location location) {
   @Override
   public void onNewRouteProgress(Location location, RouteProgress routeProgress) {
     this.routeProgress = routeProgress;
-    this.location = location;
```

Comment:
Is there a reason to remove this given that we're using `rawLocation` below?

Reply:
the location given from the onNewRouteProgress is the snapped one that was causing issues. We now directly capture the users location right when the update occurs and before we perform any actions on it.